storage-proofs-porep/src/stacked/vanilla/proof.rs

```diff
@@ -402,6 +448,26 @@ impl<'a, Tree: 'static + MerkleTreeTrait, G: 'static + Hasher> StackedDrg<'a, Tr
                                 parents_data_full,
                             ));
                         }
+
+                        // Check if a proof was detected as invalid
+                        let invalid_comm_d_coordinate = invalid_comm_d.lock().expect("failed to get lock on invalid_comm_d");
+                        if invalid_comm_d_coordinate.failure_detected {
+                            return Err(anyhow!(
+                                "Invalid comm_d detected at challenge_index {}",
+                                invalid_comm_d_coordinate.challenge_index));
+                        }
```

Comment:
It can be rewritten with the [`ensure!()` macro](https://docs.rs/anyhow/1.0.82/anyhow/macro.ensure.html):
```rust
ensure!(
    !invalid_comm_d_coordinate.failure_detected,
    "Invalid comm_d detected at challenge_index {}",
    invalid_comm_d_coordinate.challenge_index
);
```
The same is true for the ones below.